drivers/shared/executor/executor_universal_linux.go

```diff
@@ -69,125 +69,86 @@ func setCmdUser(cmd *exec.Cmd, userid string) error {
 	}
 
-	if err := cgutil.ConfigureBasicCgroups(cfg); err != nil {
-		// Log this error to help diagnose cases where nomad is run with too few
-		// permissions, but don't return an error. There is no separate check for
-		// cgroup creation permissions, so this may be the happy path.
-		e.logger.Warn("failed to create cgroup",
-			"docs", "https://www.nomadproject.io/docs/drivers/raw_exec.html#no_cgroups",
-			"error", err)
-		return nil
-	}
-	e.resConCtx.groups = cfg.Cgroups
-	return cgroups.EnterPid(cfg.Cgroups.Paths, pid)
-}
-
-func (e *UniversalExecutor) getAllPids() (map[int]*nomadPid, error) {
-	if e.resConCtx.isEmpty() {
-		return getAllPidsByScanning()
-	} else {
-		return e.resConCtx.getAllPidsByCgroup()
-	}
-}
-
-// DestroyCgroup kills all processes in the cgroup and removes the cgroup
-// configuration from the host. This function is idempotent.
-func DestroyCgroup(groups *lconfigs.Cgroup, executorPid int) error {
-	mErrs := new(multierror.Error)
-	if groups == nil {
-		return fmt.Errorf("Can't destroy: cgroup configuration empty")
-	}
-
-	// Move the executor into the global cgroup so that the task specific
-	// cgroup can be destroyed.
-	path, err := cgroups.GetInitCgroupPath("freezer")
-	if err != nil {
-		return err
-	}
-
-	if err := cgroups.EnterPid(map[string]string{"freezer": path}, executorPid); err != nil {
-		return err
-	}
-
-	// Freeze the Cgroup so that it can not continue to fork/exec.
-	groups.Resources.Freezer = lconfigs.Frozen
-	freezer := cgroupFs.FreezerGroup{}
-	if err := freezer.Set(groups.Paths[freezer.Name()], groups.Resources); err != nil {
-		return err
-	}
-
-	var procs []*os.Process
-	pids, err := cgroups.GetAllPids(groups.Paths[freezer.Name()])
-	if err != nil {
-		multierror.Append(mErrs, fmt.Errorf("error getting pids: %v", err))
-
-		// Unfreeze the cgroup.
-		groups.Resources.Freezer = lconfigs.Thawed
-		freezer := cgroupFs.FreezerGroup{}
-		if err := freezer.Set(groups.Paths[freezer.Name()], groups.Resources); err != nil {
-			multierror.Append(mErrs, fmt.Errorf("failed to unfreeze cgroup: %v", err))
-			return mErrs.ErrorOrNil()
+	lookup := func(env []string, name string) (result string) {
+		for _, s := range env {
+			if strings.HasPrefix(s, name+"=") {
+				result = strings.TrimLeft(s, name+"=")
+				return
+			}
 		}
-	}
-
-	// Kill the processes in the cgroup
-	for _, pid := range pids {
-		proc, err := os.FindProcess(pid)
-		if err != nil {
-			multierror.Append(mErrs, fmt.Errorf("error finding process %v: %v", pid, err))
-			continue
+		return
+	}
+
+	if cgutil.UseV2 {
+		// in v2 we have the definitive cgroup; create and enter it
+
+		// use the task environment variables for determining the cgroup path -
+		// not ideal but plumbing the values directly requires grpc protobuf changes
+		parent := lookup(e.commandCfg.Env, taskenv.CgroupParent)
+		allocID := lookup(e.commandCfg.Env, taskenv.AllocID)
+		task := lookup(e.commandCfg.Env, taskenv.TaskName)
```

Comment:
And those changes would be platform-specific, which would be sad to have to plumb into the protobufs if we can avoid it. It's probably worth considering whether we want to have a "platform variables" field or something in the future so we're not trying to smuggle things in thru env vars that we might not necessarily want to expose to the workload.

Reply:
> "platform variables" field 

That's a clever idea I'm definitely going to forget, so I created https://github.com/hashicorp/nomad/issues/12468